### DISCUSSION TITLE
Ubuntu 20.04 and grub-efi-amd64-signed

### DISCUSSION THREAD

**Discussion by MyreMylar** [community/community#47863] [Actions]
### Select Topic Area

Bug

### Body

Hello,

Recently we have been consistently get the following errors with grub-efi-amd64-signed on all Ubuntu 20.04 actions test runs:

```
Setting up grub-efi-amd64-signed (1.187.3~20.04.1+2.06-2ubuntu14.1) ...
mount: /var/lib/grub/esp: special device /dev/disk/by-id/scsi-14d5346542020202076e4e0de6b21a348aa1740149647cee7-part15 does not exist.
dpkg: error processing package grub-efi-amd64-signed (--configure):
 installed grub-efi-amd64-signed package post-installation script subprocess returned error exit status 32
```

Or:

```
Errors were encountered while processing:
 grub-efi-amd64-signed
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

These workflows have worked fine for a long time on 20.04 and still work fine on other versions of ubuntu (18.04 and 22.04).
We can bypass it by deliberately not installing `grub-efi-amd64-signed` but it adds an extra step to the normal setup workflow and we thought other users might also be affected. The normal setup workflow is just:

```
sudo apt-get update --fix-missing
sudo apt-get upgrade
```

And to bypass the error we have just done:

```
sudo apt-mark hold grub-efi-amd64-signed
sudo apt-get update --fix-missing
sudo apt-get upgrade
```

Instead.

Cheers,

Dan
[Pygame CE](https://github.com/pygame-community/pygame-ce/)

**Comment by JJ** [comment]
Also getting this error [here](https://github.com/JJ/paseos-por-venecia/actions/runs/4251864688/jobs/7395100767)

**Reply by jlblancoc** [reply]
And [here](https://github.com/MRPT/mrpt/actions/runs/4268827975/jobs/7431588343) now too...

**Comment by Danik2343** [comment]
i have this issue too

**Comment by r4m0n3s** [comment]
on 22.04 as well. 

```
root@a1333s05:~# apt-mark hold grub-efi-amd64-signed
grub-efi-amd64-signed was already set on hold.
root@a1333s05:~# apt-get update --fix-missing
...
Fetched 270 kB in 1s (479 kB/s)
Reading package lists... Done
root@a1333s05:~# apt-get upgrade
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
The following packages have been kept back:
  python3-software-properties software-properties-common
0 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.
2 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Do you want to continue? [Y/n] n
Abort.
```
once I confirm yes I get it back

**Comment by Danik2343** [comment]
It's fixed! Yeah!

**Comment by MyreMylar** [comment]
This seems to have returned again this week for us.

**Comment by kwagyeman** [comment]
Same

**Comment by melanciani** [comment]
Same here (although it's seems to be fine with Ubuntu `22.04.3`)

**Comment by MakoWish** [comment]
Also seeing this on 22.04.2.

**Comment by Fastidious** [comment]
This is nightmarish. Having the same problem on all 20.04 machines we are building (using MAAS). So far the only reply I have gotten is: “_It looks like you don't have an EFI partition in order to boot. Therefore, you do not need to worry about this error._” 🤯

**Comment by cipricus** [comment]
I have this after upgrade 22.04 > 23.10. No fix just yet.

**Comment by Jooosl** [comment]
I got this issue just now after updating Ubuntu 22LTS, I can't to anything anymore with apt(constantly getting the dependency and miss configuration error), putting the packages on hold also didn't help. 
Thanks for any help in advance

**Comment by nukec** [comment]
same issue here on 22LTS. i am not sure its wise to run following, since i am accessing server via SSH:
```
sudo apt-get purge grub\*
sudo apt-get install grub-efi
sudo apt-get autoremove
sudo update-grub
```

any suggestions?

**Comment by Meet6338-X** [comment]
install new 24.04 version and use this sudo apt-get upgrade in terminal it should work

**Comment by SpaceCorp-Us** [comment]
Ubuntu 22.04 - Same Error as Above

username@mySystem:~$ sudo apt-get upgrade
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
1 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Do you want to continue? [Y/n] y
Setting up shim-signed (1.51.4+15.8-0ubuntu1) ...
/var/lib/dpkg/info/shim-signed.postinst: 165: /usr/lib/grub/grub-multi-install: not found
dpkg: error processing package shim-signed (--configure):
 installed shim-signed package post-installation script subprocess returned error exit status 127
Errors were encountered while processing:
 shim-signed
E: Sub-process /usr/bin/dpkg returned an error code (1)

Please let Us know How to Fix this... Thanks !

**Comment by Retalak** [comment]
Same thing happened to me on Ubuntu 24.04 LTS with software raid.

```
Personalities : [raid1] [raid6] [raid5] [raid4] [raid0] [raid10]
md125 : active raid5 sdb4[1] sdc4[2] sdd4[3] sda4[0]
      4187136 blocks super 1.2 level 5, 512k chunk, algorithm 2 [4/4] [UUUU]

md126 : active raid5 sdb5[1] sdc5[2] sdd5[3] sda5[0]
      5837076480 blocks super 1.2 level 5, 512k chunk, algorithm 2 [4/4] [UUUU]
      bitmap: 6/15 pages [24KB], 65536KB chunk

md127 : active raid1 sda3[0] sdd3[3] sdc3[2] sdb3[1]
      4189184 blocks super 1.2 [4/4] [UUUU]

md2 : active raid1 sdb2[1] sdd2[3] sdc2[2] sda2[0]
      1048512 blocks [4/4] [UUUU]

md1 : active raid1 sdd1[3] sdb1[1] sdc1[2] sda1[0]
      1048512 blocks [4/4] [UUUU]

unused devices: <none>
```

Every time I would run apt upgrade/install/autoremove I would get the GUI configuration screen saying "The GRUB boot loader was previously installed to a disk that is no longer present, or whose unique identifier has changed for some reason."

My machine is booting fine regardless of the error. I wound up just selecting none of the 4 raid 1 partitions and confirming that I did not want to configure grub-efi-amd64-signed. Not sure that it is even a necessary package, but I am not going to find out by uninstalling it. These are the grub packages I have installed:

```
grub-common 2.12-1ubuntu7.1 [Ubuntu/noble main]
├── is installed
└── GRand Unified Bootloader (common files)

grub-efi 2.12-1ubuntu7.1 [Ubuntu/noble main]
├── is installed
└── GRand Unified Bootloader, version 2 (dummy package)

grub-efi-amd64 2.12-1ubuntu7.1 [Ubuntu/noble main]
├── is installed
└── GRand Unified Bootloader, version 2 (EFI-AMD64 version)

grub-efi-amd64-bin 2.12-1ubuntu7.1 [Ubuntu/noble main]
├── is installed
└── GRand Unified Bootloader, version 2 (EFI-AMD64 modules)

grub-efi-amd64-signed 1.202.2+2.12-1ubuntu7.1 [Ubuntu/noble main]
├── is installed
└── GRand Unified Bootloader, version 2 (EFI-AMD64 version, signed)

grub-emu 2.12-1ubuntu7.1 [Ubuntu/noble universe]
├── is installed
└── GRand Unified Bootloader, version 2 (emulated version)

grub2-common 2.12-1ubuntu7.1 [Ubuntu/noble main]
├── is installed
└── GRand Unified Bootloader (common files for version 2)
```

**Comment by koenr** [comment]
We are suffering from this problem as well (on 22.04 and 24.04). We distribute a script to our machines to be able to quickly solve it manually, containing

killall dpkg grub-multi-inst
dpkg-reconfigure shim-signed
update-grub

The main problem for us is that all subsequent unattended upgrades and scripted software installs with apt (we use saltstack) don't run anymore if the machine is stuck in this state. Laptops in this state drain their batteries in a couple of hours and that is usually why we see the problematic devices.
